chain/index/interface.go

```diff
@@ -0,0 +1,31 @@
+package index
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+)
+
+// MsgInfo is the Message metadata the index tracks.
+type MsgInfo struct {
+	// the message this record refers to
+	Message cid.Cid
+	// the epoch whre this message was executed
+	Epoch abi.ChainEpoch
+	// the tipset where this messages executed
+	Tipset types.TipSetKey
+	// the first block in the tipset where the message was executed
+	Block cid.Cid
+	// the index of the message in the block
+	Index int
```

Comment:
I think you can remove the Block and instead have the canonicalized execution index (which is the index at which the receipt will be stored in the AMT).

Reply:
ok, I will see whether I can get rid of it when I get to implementation details.

Reply:
Yeah, we don't need it; dropped it from the interface.